### PR TITLE
IsIdentifiable patterns can be case sensitive

### PR DESCRIPTION
**Pull request by howff**
## Proposed Changes

The IfPatternCaseSensitive has been removed, replaced by IfPatternIsCaseSensitive:bool which *must* precede IfPattern.
The Rules yaml files have been updated appropriately.

See [CHANGELOG](https://github.com/SMI/SmiServices/blob/master/CHANGELOG.md) diff.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [X ] Bugfix (non-breaking change which fixes an issue)
- [X ] New Feature (non-breaking change which adds functionality)
- [X ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [X ] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [ X] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [ X] Updated any relevant API documentation
- [ ] Created or updated any tests if relevant
- [ X] Accurately updated the [CHANGELOG](https://github.com/SMI/SmiServices/blob/master/CHANGELOG.md)
  - NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [ ] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [ X] Requested a review by one of the repository maintainers

## Issues

If relevant, tag any issues that are *expected* to be resolved with this PR. E.g.:

- Closes #\<issue-number>
- ...

### CODE REVIEW

File: src/microservices/Microservices.IsIdentifiable/Rules/IsIdentifiableRule.cs [hunk 1]
```diff
@@ -43,17 +42,13 @@ public class IsIdentifiableRule : ICustomRule
...
-        /// The Regex pattern which should be used to match values with
+        /// Whether the IfPattern match is case sensitive or not (default is false)
         /// </summary>
-        public string IfPatternCaseSensitive
-        {
-            get => _ifPattern?.ToString();
-            set => _ifPattern = value == null ? null : new Regex(value, RegexOptions.Compiled);
-        }
-
+        public bool IfPatternIsCaseSensitive  { get; set; }
```
Comment (tznind) [inline src/microservices/Microservices.IsIdentifiable/Rules/IsIdentifiableRule.cs:51]: Just `CaseSensitive` would probably be enough for a name?  We should be able to use a backing private variable _caseSensitive then on `set` we can change `_ifPattern` to a new Regex? that way we can support changing the sensitivity on the object and not have to worry about order of property setting in yaml deserialization (which may not nessesarily match how it is read from the file necessarily?  Also would be unexpected with the current codebase that the set does nothing unless youa lso change the IfPattern immediately afterwards?

File: src/microservices/Microservices.IsIdentifiable/Rules/WhiteListRule.cs [hunk 2]
```diff
@@ -50,35 +48,25 @@ public class WhiteListRule : ICustomRule
         public string IfPattern
         {
             get => _ifPattern?.ToString();
-            set => _ifPattern = value == null ? null : new Regex(value, RegexOptions.IgnoreCase | RegexOptions.Compiled);
+            set => _ifPattern = value == null ? null : new Regex(value, (IfPatternIsCaseSensitive ? 0 : RegexOptions.IgnoreCase) | RegexOptions.Compiled);
```
Comment (tznind) [inline src/microservices/Microservices.IsIdentifiable/Rules/WhiteListRule.cs:51]: What is the difference between WhiteListRule and IsIdentifiableRule? do we need both?

Comment (howff) [reply src/microservices/Microservices.IsIdentifiable/Rules/WhiteListRule.cs:51]: WhiteListRules have two patterns, one to match the whole string and one to match the substring part which previously matched a rule.

Comment (tznind) [reply src/microservices/Microservices.IsIdentifiable/Rules/WhiteListRule.cs:51]: Ah ok cool, can we cut down on code duplication by inheriting one from the other? (just an idea)

File: src/microservices/Microservices.IsIdentifiable/Rules/IsIdentifiableRule.cs (hunk 1 above)
Comment (tznind) [reply src/microservices/Microservices.IsIdentifiable/Rules/IsIdentifiableRule.cs:51]: Ok that makes sense.  I can make the change if you want and add some unit tests, its a nice self contained class so should be easy to test? Definetly think its worth the effort, if a class has a public set it should work toggling it at any state of the object (or throw appropriate exceptions)

**Pull request merged**